akka-stream/src/main/scala/akka/stream/impl/io/IOSinks.scala

```diff
@@ -66,3 +74,163 @@ private[akka] final class OutputStreamSink(createOutput: () ⇒ OutputStream, va
   override def withAttributes(attr: Attributes): Module =
     new OutputStreamSink(createOutput, attr, amendShape(attr))
 }
+
+/**
+ * INTERNAL API
+ * Creates simple synchronous (Java 6 compatible) Sink which provide all incoming element on read request via
+ * InputStream.
+ */
+private[akka] class InputStreamSink(val timeout: FiniteDuration,
+                                    val attributes: Attributes, shape: SinkShape[ByteString])
+  extends SinkModule[ByteString, (InputStream, Future[Long])](shape) {
```

Comment:
do we really need the Future[Long]? If we do, then I recommend creating an explicit SinkInputStream interface that exposes that Future, then there is no need to return a pair.